### ISSUE TITLE
Add theme to permalink

### ISSUE THREAD

**Issue by elemoine** [Geoportail-Luxembourg/geoportailv3#400]
As opposed to other states the name of the theme should be in the URL's path.

Restant c2c 01/04/2015 1j

**Comment by elemoine** [comment]
@jaykayone could you please confirm that "theme" permalinks should look like as in version 2? E.g. http://eau.geoportail.lu/?zoom=2&X=100000&Y=80000.

**Comment by elemoine** [comment]
http://map.geoportail.lu/eau?zoom=2&X=100000&Y=80000 also works. I guess this is the one the application should support, right?

**Comment by elemoine** [comment]
New comment: in c2cgeoportal "theme" permalinks look like this: http://example.com/theme/eau. Note the word "theme" in the URL path. I guess this won't work for geoportailv3, right?

**Comment by jaykayone** [comment]
eau.geoportail.lu is a rewrite to map.geoportail.lu/eau

the second one is the one to implementthe rest is done with rewrites

the /theme can be ok for us in v3 but with ?version=2 we have to support the /${theme}

**Comment by elemoine** [comment]
Thanks Jeff. It makes sense. But we will need server-side changes to support `/{theme}` URLs. 

Another question: do we also store the current theme in the local storage? 

Thanks.

**Comment by jaykayone** [comment]
yes, if someone comes through a rewritten subdomain it will be overloaded

**Comment by elemoine** [comment]
> yes, if someone comes through a rewritten subdomain it will be overloaded

Ok.

Currently, if we find parameters in the URL, we do not look at the local storage. For example, if the URL is `/?zoom=15` then we use zoom 15 and default values for X and Y.

I wonder if the same should apply for themes. For example, if the URL is `/theme/eau` (or `/eau`) without any query parameters, should we use default values for X and Y, or should we use values from the local storage (if any)?

In other words, are themes handled differently from the other application states?

I am not sure my questions are clear. Maybe something to discuss at the next telco.

**Comment by jaykayone** [comment]
this is a good question
I think it should be discussed at the next telco

**Comment by jaykayone** [comment]
Decision: 
- No storage of the theme in the localstorage
- Different handling, which means that if only the theme is present we take the rest from localstorage

**Comment by elemoine** [comment]
"question" tag removed.

**Comment by pgiraud** [comment]
This morning, during the meeting, while we were talking about whether to get the theme from storage or not, I had a thought.
What about asking the user whether he/she wants to restore the previous context?
A good example of such a functionality is used in http://geojson.io in which the user is prompted "recover your map from last time you edited?" in case he/she used the service before.

**Comment by jaykayone** [comment]
good idea to keep in mind for later on.
Currently we should focus on finishing a first version
